### ISSUE TITLE
noUndeclaredDependencies: devDependencies restriction is not applied to type-only imports

### ISSUE THREAD

**Issue by bquintana** [biomejs/biome#11388]
### Environment information

Biome 2.5.8, macOS. Reproduced with a standalone 6-line project (below).

### Description

`correctness/noUndeclaredDependencies` supports restricting which files may import `devDependencies`, via the `devDependencies` option's glob-array form. That restriction is **not applied to type-only imports**: `import type { X } from "pkg"` is exempt, while `import { y } from "pkg"` in the same file, for the same package, is reported.

The absent-from-every-manifest-field case *is* reported for type-only imports, so this is specifically about the `devDependencies` restriction.

I'd like the option to have the restriction apply to type-only imports too.

### Reproduction

`package.json`:

```json
{
  "name": "repro",
  "version": "1.0.0",
  "dependencies": {},
  "devDependencies": { "lodash": "*" }
}
```

`biome.json`:

```json
{
  "$schema": "https://biomejs.dev/schemas/2.5.8/schema.json",
  "root": true,
  "linter": {
    "enabled": true,
    "rules": {
      "recommended": false,
      "correctness": {
        "noUndeclaredDependencies": {
          "level": "error",
          "options": { "devDependencies": ["**/*.test.ts"] }
        }
      }
    }
  }
}
```

`src/a.ts`:

```ts
import type { LoDashStatic } from "lodash";
import { merge } from "lodash";

export type A = LoDashStatic;
export const m = merge;
```

### Expected

`src/a.ts` is not a `**/*.test.ts`, so both imports of the devDependency `lodash` are reported.

### Actual

Only line 2 is reported. Line 1 is silently accepted.

```
src/a.ts:2:23 lint/correctness/noUndeclaredDependencies ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  × Dependency lodash isn't specified in package.json.

  i lodash is part of your devDependencies, but it's not intended to be used in this file.
```

Same behaviour for a package whose types come from `@types/*`: with `@types/geojson` in `devDependencies`, `import type { Feature } from "geojson"` is exempt.

### Why this matters

For a package that **publishes raw TypeScript source** — `exports` pointing at `./src/*.ts` rather than at built `.js` + `.d.ts` — an undeclared type-only import is a real defect, not a dev-time concern. Consumers do not install `devDependencies`, but they *do* typecheck against the shipped source, so the missing package surfaces as a type error in the consumer's build.

We hit this on a real monorepo of published packages. The rule with the glob allowlist caught every value-import case exactly, which was genuinely useful; the type-only ones were the only class it did not surface, and they fail downstream the same way — just at `tsc` rather than at runtime.

### Possible shape

An opt-in so the existing behaviour stays the default, e.g.:

```json
"options": {
  "devDependencies": ["**/*.test.ts"],
  "checkTypeImports": true
}
```

Happy to be pointed at a different mechanism if one already covers this — I checked `publint` and `@arethetypeswrong/cli` and neither analyses the import graph for declaration coverage.

**Comment by biome-cookie** [comment]
Hello @bquintana, please provide a minimal reproduction. You can use one of the following options:

- Provide a link to [our playground](https://biomejs.dev/playground), if it's applicable.
- Provide a link to GitHub repository. To easily create a reproduction, you can use our interactive CLI via `npm create @biomejs/biome-reproduction`

Issues marked with `S-Needs repro` will be **closed** if they have **no activity within 3 days**.

**Comment by biome-cookie** [comment]
This issue has been automatically closed because it has been inactive for more than 3 days and it does not have a minimal reproduction.
